react/features/device-selection/components/VideoInputPreview.js

```diff
@@ -0,0 +1,236 @@
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentDidMount() {
+        this._attachTrack(this.props.track);
+    }
+
+    /**
+     * Only update when the deviceId has changed. This component is
+     * somewhat black-boxed from React's rendering so lib-jitsi-meet
+     * can instead handle updating of the video preview, which takes
+     * browser differences into consideration. For example, temasys's
+     * video object must be visible to update the displayed track,
+     * but React's re-rendering could potentially remove the
+     * video object from the page.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    shouldComponentUpdate(nextProps) {
+        if (nextProps.track !== this.props.track) {
+            this._attachTrack(
+                nextProps.track,
+                this.props.track
+            );
+        }
+
+        return false;
+    }
+
+    /**
+     * Remove any existing associations between the current previewed track
+     * and the component's video element.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentWillUnmount() {
+        this._detachTrack(this.props.track);
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        return (
+            <div
+                className = 'video-input-preview'
+                ref = { this._setRootEl }>
+                <video
+                    autoPlay = { true }
+                    className = 'video-input-preview-display flipVideoX'
+                    ref = { this._setVideoEl } />
+                <div className = 'video-input-preview-muted'>
+                    { this.props.t('videothumbnail.muted') }
+                </div>
+            </div>
+        );
+    }
+
+    /**
+     * Adds or removes a class to the component's parent node to indicate
+     * mute status.
+     *
+     * @param {bool} shouldShow - True if the mute class should be added and
+     * false if the class should be removed.
+     * @returns {void}
+     */
+    _showMuteOverlay(shouldShow) {
+        const rootEl = this._getRootEl();
+
+        if (shouldShow) {
+            rootEl.classList.add(videoMuteClass);
+        } else {
+            rootEl.classList.remove(videoMuteClass);
+        }
+    }
+
+    /**
+     * Calls into the passed in track to associate the track with the
+     * component's video element and render video. Also sets the instance
+     * variable for the video element as the element the track attached to,
+     * which could be an Object if on a temasys supported browser.
+     *
+     * @param {JitsiLocalTrack} newTrack - The library's track model which
+     * will be displayed.
+     * @param {JitsiLocalTrack} oldTrack - The previously displayed track
+     * model which should stop displaying.
+     * @returns {void}
+     * @private
+     */
+    _attachTrack(newTrack, oldTrack) {
```

Comment:
I think this function should just take the new track and the caller should be responsible from detaching the previous one.